AsyncDisplayKit/TextKit/ASTextKitRenderer.mm

```diff
@@ -161,15 +188,13 @@ - (void)_calculateSize
   
   [[self truncater] truncate];
   
+  CGRect constrainedRect = {CGPointZero, _constrainedSize};
+  __block CGRect boundingRect;
+
   // Force glyph generation and layout, which may not have happened yet (and isn't triggered by
   // -usedRectForTextContainer:).
   [[self context] performBlockWithLockedTextKitComponents:^(NSLayoutManager *layoutManager, NSTextStorage *textStorage, NSTextContainer *textContainer) {
     [layoutManager ensureLayoutForTextContainer:textContainer];
-  }];
-  
-  CGRect constrainedRect = {CGPointZero, _constrainedSize};
-  __block CGRect boundingRect;
-  [[self context] performBlockWithLockedTextKitComponents:^(NSLayoutManager *layoutManager, NSTextStorage *textStorage, NSTextContainer *textContainer) {
```

Comment:
We were calling `performBlockWithLock…` twice in a row, without doing any work in-between!